src/mcp_server_uyuni/server.py

```diff
@@ -37,6 +38,30 @@ class ActivationKeySchema(BaseModel):
 
 logger = get_logger(log_file=CONFIG["UYUNI_MCP_LOG_FILE_PATH"], transport=CONFIG["UYUNI_MCP_TRANSPORT"])
 
+class AuthTokenMiddleware(Middleware):
+    async def on_call_tool(self, ctx: MiddlewareContext, call_next):
+        """
+        Extracts the JWT token from the Authorization header (if present)
+        and injects it into the context state for other tools to use.
+        """
+        fastmcp_ctx = ctx.fastmcp_context
+        auth_header = fastmcp_ctx.request_context.request.headers['authorization']
+        token = None
+        if auth_header:
```

Comment:
could we add a config variable that controls if auth is expected for mlm? Could be we have a token for the mcp authorization but mlm does not yet support it..